include/dlaf/matrix_const.tpp

```diff
@@ -77,7 +77,7 @@ hpx::shared_future<Tile<const T, device>> Matrix<const T, device>::read(
 }
 
 template <class T, Device device>
-Matrix<const T, device>::Matrix(Distribution&& distribution,
+Matrix<const T, device>::Matrix(Distribution distribution,
                                 std::vector<hpx::future<TileType>>&& tile_futures,
                                 std::vector<hpx::shared_future<ConstTileType>>&& tile_shared_futures)
```

Comment:
Not sure if this is a fly-by change but you can take the remaining arguments by value (i.e. std::vector<...>) and move them the same way you did for `Distribution`. If the class is going to take ownership of the objects (which it does), declaring them by value may have some optimization benefits and covers the const& case (i.e when you anyway have to make a copy). There was a talk about that by Chandler Carruth sometime ago: https://www.youtube.com/watch?v=eR34r7HOU14&feature=youtu.be

Reply:
Yes, I agree with you. It is something I learnt months ago, but unfortunately after I implemented part of the structures in DLAF. I can change them here too if you agree.

Reply:
While I was looking at it, I found out that it was not even needed anymore. So, I took the chance and moved the majority of these Matrix changes to a dedicated PR.